ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -691,12 +694,22 @@ extension BrowserViewController: WKNavigationDelegate {
       let response = response as? HTTPURLResponse
     {
       let internalUrl = InternalURL(responseURL)
-      let isErrorPage = internalUrl?.isErrorPage == true || response.statusCode >= 400
-      let isSessionRestore = internalUrl?.isSessionRestore == true
 
-      if isErrorPage || isSessionRestore {
-        tab.shouldNotifyAdsServiceTabDidChange = false
-        tab.shouldNotifyAdsServiceTabContentDidChange = false
+      tab.rewardsReportingState.isErrorPage = internalUrl?.isErrorPage == true
+      if !tab.rewardsReportingState.isErrorPage {
+        let kClientErrorResponseCodeClass = 4
+        let kServerErrorResponseCodeClass = 5
```

Comment:
nit: `kHttpClientErrorResponseCodeClass` and `kHttpServerErrorResponseCodeClass`

Reply:
Fixed